File: javascript/module2/solutions/solution4.js
function getTotalDamage(baseDamage, ringBonus, wandBonus, isDoubleDamage) {
  const minRingBonusToCauseDamage = 5;
  const minWandBonusToCauseDamage = 3;

  if (
    ringBonus < minRingBonusToCauseDamage ||
    wandBonus < minWandBonusToCauseDamage
  ) {
    return 0;
  }

  let result = baseDamage * ringBonus * wandBonus;

  if (isDoubleDamage) {
    result *= 2;
  }

  return result;
}

File: csharp/module2/solutions/solution4.cs
public class DamageCalculator
{
    private const int MinRingBonusToCauseDamage = 5;
    private const int MinWandBonusToCauseDamage = 3;

    public int GetTotalDamage(int baseDamage, int ringBonus, int wandBonus, bool isDoubleDamage)
    {
        if (ringBonus < MinRingBonusToCauseDamage || wandBonus < MinWandBonusToCauseDamage)
        {
            return 0;
        }

        int result = baseDamage * ringBonus * wandBonus;

        if (isDoubleDamage)
        {
            result *= 2;
        }

        return result;
    }
}
